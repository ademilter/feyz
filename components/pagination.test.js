import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Pagination from './pagination'

describe('Pagination', () => {
  test('should have render.', () => {
    const { asFragment } = render(<Pagination totalData={30} />)

    expect(asFragment()).toMatchSnapshot()
    expect(screen.queryByTestId('pagination').childElementCount).toBe(3)
  })

  test('should have render with tag.', () => {
    const { asFragment } = render(<Pagination totalData={30} tag="test" />)

    expect(asFragment()).toMatchSnapshot()
    expect(screen.queryByTestId('pagination').childElementCount).toBe(3)
    expect(screen.getByText(/3/i)).toHaveAttribute('href', '/tag/test/3')
  })
})
