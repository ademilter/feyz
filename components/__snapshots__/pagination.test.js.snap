// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should have render with tag. 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="pagination"
  >
    <a
      href="/tag/test"
    >
      1
    </a>
    <a
      href="/tag/test/2"
    >
      2
    </a>
    <a
      href="/tag/test/3"
    >
      3
    </a>
  </div>
</DocumentFragment>
`;

exports[`Pagination should have render. 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="pagination"
  >
    <a
      href="/"
    >
      1
    </a>
    <a
      href="/2"
    >
      2
    </a>
    <a
      href="/3"
    >
      3
    </a>
  </div>
</DocumentFragment>
`;
